ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/ScheduledProtocolSpec.java

```diff
@@ -25,10 +27,24 @@ public interface ScheduledProtocolSpec {
 
   boolean isOnMilestoneBoundary(ProcessableBlockHeader header);
 
-  long milestone();
+  // long milestone();
+  Hardfork fork();
 
   ProtocolSpec spec();
 
+  public record Hardfork(String name, long milestone) implements Comparable<Hardfork> {
```

Comment:
Not sure we _need_ it since it is a readability improvement, but it encapsulates transition points and allows them to be compared, which is used by Engine APIs.

Reply:
I like how it reads in the client code for sure. I haven't thought through the readability of using ScheduledProtocolSpec instead. 

My concern is that there is a potential comprehension/complexity issue with introducing yet another Type that represents something about a fork... there's already milestone, ProtocolSpec, ScheduledProtocolSpec, ForkSpec (for bft)